packages/gatsby-plugin-sharp/src/utils.js

```diff
@@ -98,77 +117,108 @@ export const calculateImageSizes = ({
   outputPixelDensities,
   srcSetBreakpoints,
 }) => {
-  // Sort, dedupe and ensure there's a 1
-  const densities = Array.from(new Set([1, ...outputPixelDensities])).sort()
+  // check that all dimensions provided are positive
+  const userDimensions = { width, maxWidth, height, maxHeight }
+  const erroneousUserDimensions = Object.entries(userDimensions).filter(
+    ([_, size]) => size < 1
+  )
+  if (erroneousUserDimensions.length) {
+    throw new Error(
+      `Specified dimensions for images must be positive numbers (> 0). Problem dimensions you have are ${erroneousUserDimensions
+        .map(dim => dim.join(`: `))
+        .join(`, `)}`
+    )
+  }
+
   let sizes
+  const aspectRatio = imgDimensions.width / imgDimensions.height
+  // Sort, dedupe and ensure there's a 1
+  const densities = Array.from(
+    new Set([1, ...(outputPixelDensities || DEFAULT_PIXEL_DENSITIES)])
+  ).sort()
+
   if (layout === `fixed`) {
-    // FIXED
-    // if no width is passed, we need to resize the image based on the passed height
-    const fixedDimension = width === undefined ? `height` : `width`
+    checkIgnoredParameters(`fixed`, { maxWidth, maxHeight }, file.absolutePath)
```

Comment:
I would call this function differently, it's doing more than checking
```suggestion
    warnIgnoredParametersUsage(`fixed`, { maxWidth, maxHeight }, file.absolutePath)
```